README.md

```diff
@@ -1 +1,338 @@
+
+## Table of Contents
+
+- [Purpose](#purpose)
+- [Features](#features)
+- [Installation](#installation)
+- [Usage](#usage)
+- [API Documentation](#apidocumentation)
+- [Contributing](#contributing)
+
+
+## Purpose
+
+1.	Connect customers with talented local cooks who offer a diverse selection of homemade dishes.
+2.	Support local cooks by providing them a platform to showcase their culinary creations and skills.
+
+
+
+## Features
+
+1. Cooks can create detailed profiles showcasing their restaurant and culinary specialties.
+2. Cooks can upload a variety of homemade dishes to their menus.
+3. Cooks and customers can track the status of orders, from preparation to delivery.
+4. Customers can browse and explore a diverse range of homemade dishes.
+5. Customers can add favorite dishes in their profile.
+6. Customers can provide ratings for dishes they've ordered.
+
+## Installation
+
+Setting up the Cookiz app is a breeze. Follow these steps:
+
+- Clone the repository to your local machine using 'git clone repo-url'.
+- Execute 'yarn install' to install the necessary dependencies.
+
+### To start the server
+
+Run the command `yarn start`. This will install all the required dependencies.
+
+### To run the tests
+
+Execute the command `yarn test`. Ensure that both the server and the database are up and running.
+
+### To install packages
+
+When you run `yarn add PACKAGE`, the package will be automatically installed in Docker as well. If you encounter any issues, stop the server first using `yarn run stop`, then rebuild the Docker image with `yarn run build` before starting again.
+
+## Usage
+
+Interact with the backend server by making HTTP requests to the following endpoints.  
+The base URL for all endpoints is: http://localhost:3000/api
+
+**Endpoints**  
+
+**1. Sign Up**  
+Description: Creates a new user account  
+Endpoint: `/auth/signup`  
+Method: POST  
+Request Body:
+```json
+{
+    "firstname": "Taylor",
+    "lastname": "Smith",
+    "email": "taylor.smith@email.com",
+    "password": "Password1",
+    "role": ""
+}
+```
+
+**2. Login**  
+Description: Authenticates the user and generates an access token for subsequent requests  
+Endpoint: `/auth/login`  
+Method: POST  
+Request Body:
+```json
+{
+    "email": "taylor.smith@email.com",
+    "password": "Password1"
+}
+```
+
+**3. Password Reset**  
+Description: Initiates the password reset process for a user  
+Endpoint: `/auth/passwordReset`  
+Method: POST  
+Request Body:
+```json
+{
+    "email": "taylor.smith@email.com",
+    "newPassword": "Newpassword1"
+}
+```
+**4. Logout**  
+Description: Logs out the currently logged-in user and invalidates the access token  
+Endpoint: `/auth/logout`  
+Method: POST  
+
+**5. Decide User Role**  
+Description: Allows the system to assign a role to a user, based on certain criteria or user preferences  
+Endpoint: `/role`  
+Method: PUT  
+Request Body:
+```json
+{
+    "providerId": "",
```

Comment:
Just delete this cuz we no longer need it, We should depend on the userId that we are going to extract from the token, this is what we need. In the branch of refactoring googleAuth I fixed it, so we no longer need line 109